Document/0x05c-Reverse-Engineering-and-Tampering.md

```diff
@@ -1723,96 +1723,89 @@ import android.os.Bundle;
-
-    b(sg.vantagepoint.uncrackable1.MainActivity a0)
-    {
-        this.a = a0;
-        super();
-    }
-
-    public void onClick(android.content.DialogInterface a0, int i)
-    {
-        System.exit(0);
-    }
-}
 ```
 
-It just exits the app. Now intercept it with Frida to prevent the app from exiting after root detection:
+Notice the "Root detected" message in the `onCreate` method and the various methods called in the preceding `if`-statement (which perform the actual root checks). Also note the "This is unacceptable..." message from the first method of the class, `private void a`. Obviously, this method displays the dialog box. There is an `alertDialog.onClickListener` callback set in the `setButton` method call, which closes the application via `System.exit(0)` after successful root detection. With Frida, you can prevent the app from exiting by hooking the MainActivity.a method or the callback inside it. The example below shows how you can hook MainActivity.a and prevent it from ending the application. 
 
-```java
+```javascript
 setImmediate(function() { //prevent timeout
     console.log("[*] Starting script");
 
     Java.perform(function() {
-      bClass = Java.use("sg.vantagepoint.uncrackable1.b");
-      bClass.onClick.implementation = function(v) {
-         console.log("[*] onClick called");
+      var mainActivity = Java.use("sg.vantagepoint.uncrackable1.MainActivity");
+      mainActivity.a.implementation = function(v) {
+         console.log("[*] MainActivity.a called");
       };
-      console.log("[*] onClick handler modified");
+      console.log("[*] MainActivity.a modified");
 
     });
 });
 ```
 
-Wrap your code in the function `setImmediate` to prevent timeouts (you may or may not need to do this), then call `Java.perform` to use Frida's methods for dealing with Java. Afterwards retrieve a wrapper for the class that implements the `OnClickListener` interface and overwrite its `onClick` method. Unlike the original, the new version of `onClick` just writes console output and *doesn't exit the app*. If you inject your version of this method via Frida, the app should not exit when you click the "OK" dialog button.
+Wrap your code in the function `setImmediate` to prevent timeouts (you may or may not need to do this), then call `Java.perform` to use Frida's methods for dealing with Java. Afterwards retrieve a wrapper for `MainActivity` class and overwrite its `a` method. Unlike the original, the new version of `a` just writes console output and *doesn't exit the app*. An alternative solution is to hook onClick method of the OnClickListener interface. You can overwrite the onClick method and prevent it from ending the application with `System.exit(0)` call. If you want to inject your own Frida  script, the script should either disable the `AlertDialog` entirely or change the behavior of the onClick method so the app does not exit when you click the "OK" dialog button.
```

Comment:
Added backticks to be consistent with the style guide

```suggestion
Wrap your code in the function `setImmediate` to prevent timeouts (you may or may not need to do this), then call `Java.perform` to use Frida's methods for dealing with Java. Afterwards retrieve a wrapper for `MainActivity` class and overwrite its `a` method. Unlike the original, the new version of `a` just writes console output and doesn't exit the app. An alternative solution is to hook `onClick` method of the `OnClickListener` interface. You can overwrite the `onClick` method and prevent it from ending the application with the `System.exit` call. If you want to inject your own Frida script, it should either disable the `AlertDialog` entirely or change the behavior of the `onClick` method so the app does not exit when you click "OK".
```